### DISCUSSION TITLE
Bidirectional DShot

### DISCUSSION THREAD

**Discussion by symonb** [iNavFlight/inav#8658] [Ideas]
I've implemented BDShot for my project (FC based on STM32 F405) and I think it can be useful for implementation in iNav. I created a separate repository for BDShot and RPM filter but it is the same as the version that I've "tested". I've tried to comment on everything and kept it as simple as possible. At this moment I don't have time to dig into iNav code and implement it myself but maybe later... Anyway, I hope it will be useful: https://github.com/symonb/Bidirectional-DSHOT-and-RPM-Filter.git
So far I've only tested that drone flying with both features turned on. I've got no idea about flight performance couse my drone is too far from a well-flying machine.

**Comment by Alex-TV** [comment]
I wonder how long it takes to include this in the main repository.

**Reply by lida2003** [reply]
Hmm... just rember the feature long time ago. https://github.com/iNavFlight/inav/issues/5549

**Reply by sensei-hacker** [reply]
> I wonder how long it takes to include this in the main repository.

First, someone would have to write an implementation for INAV. How long that takes - well it's been a few years and nobody has started yet.

I don't plan on working on that this week. Do you? That's kinda where it's been.
